tests/candidates/test_matchers.py

```diff
@@ -173,3 +196,231 @@ def test_inverse(doc_setup):
+    stemmer = PorterStemmer()
+    matcher = DictionaryMatch(d=["is"], stemmer=stemmer)
+    # _stem(w) should return a word stem.
+    assert matcher._stem("caresses") == "caress"
+
+    stemmer.stem = Mock(
+        side_effect=UnicodeDecodeError("dummycodec", b"\x00\x00", 1, 2, "Dummy  !")
+    )
+    matcher = DictionaryMatch(d=["is"], stemmer=stemmer)
+    # _stem(w) should return w as stemmer.stem raises UnicodeDecodeError.
+    assert matcher._stem("caresses") == "caresses"
+
+
+def test_lambda_function_matcher(doc_setup):
+    """Test DictionaryMatch matcher."""
+    doc = doc_setup
+    space = MentionNgrams(n_min=1, n_max=1)
+
+    # Test with a lambda function
+    matcher = LambdaFunctionMatcher(func=lambda x: True)
+    assert set(tc.get_span() for tc in matcher.apply(space.apply(doc))) == {
+        "This",
+        "is",
+        "apple",
+    }
+
+    # Test if matcher raises an error when _f is given non-TemporarySpanMention
+    with pytest.raises(ValueError):
+        list(matcher.apply(doc.sentences[0].words))
+
+    # Test if an error raised when a func is not provided.
+    with pytest.raises(Exception):
+        LambdaFunctionMatcher()
+
+
+def test_regex_match(doc_setup):
+    """Test RegexMatch matcher."""
+    doc = doc_setup
+    space = MentionNgrams(n_min=1, n_max=2)
+
+    # a wrong option name should raise an excetiopn
+    with pytest.raises(Exception):
+        RegexMatchSpan(regex=r"apple")
+
+    # Test if matcher raises an error when _f is given non-TemporarySpanMention
+    matcher = RegexMatchSpan(rgx=r"apple")
+    with pytest.raises(ValueError):
+        list(matcher.apply(doc.sentences[0].words))
+
+    matcher = RegexMatchEach(rgx=r"apple")
+    with pytest.raises(ValueError):
+        list(matcher.apply(doc.sentences[0].words))
+
+    # Test if RegexMatchEach works as expected.
+    assert set(tc.get_span() for tc in matcher.apply(space.apply(doc))) == {"apple"}
+
+    # Test ignore_case option
+    matcher = RegexMatchEach(rgx=r"Apple", ignore_case=False)
+    assert list(matcher.apply(space.apply(doc))) == []
+
+
+def test_ner_matchers():
+    """Test different ner type matchers."""
+    # Set up a document
+    doc = Document(id=1, name="test", stable_id="1::document:0:0")
+    doc.text = " ".join(
+        [
+            "Tim Cook was born in USA in 1960.",
+            "He is the CEO of Apple.",
+            "He sold 100 million of iPhone.",
+        ]
+    )
+    lingual_parser = SpacyParser("en")
+    for parts in lingual_parser.split_sentences(doc.text):
+        parts["document"] = doc
+        Sentence(**parts)
+    # Manually attach ner_tags as the result from spacy may fluctuate.
+    doc.sentences[0].ner_tags = [
+        "PERSON",
+        "PERSON",
+        "O",
+        "O",
+        "O",
+        "GPE",
+        "O",
+        "DATE",
+        "O",
+    ]
+    doc.sentences[1].ner_tags = ["O", "O", "O", "O", "O", "ORG", "O"]
+    # TODO: replace "NUMBER" with "CARDINAL" (#473)
```

Comment:
Let's make this in another PR.
Maybe check this depends on SpaCy version?

Reply:
I'd fix this in another PR (which will fix #473).

> Maybe check this depends on SpaCy version?

What do you mean by this?